core/ArchiveProcessor/Loader.php

```diff
@@ -113,6 +114,12 @@ private function prepareArchiveImpl($pluginName)
             }
         }
 
+        // invalidate existing archives before we start archiving in case data was tracked in the past. if the archive is
+        // made invalid, we will correctly re-archive below.
+        if ($this->invalidateBeforeArchiving) {
+            $this->invalidatedReportsIfNeeded();
+        }
```

Comment:
Note: Invalidating before we look for an archive, otherwise, the invalidation won't happen.

Reply:
@tsteur I think it's required for browser archiving to support the "tracking data in the past" feature. Otherwise, the invalidations that get queued in the option table never get applied. We could add `isBrowserArchivingEnabled` as a safety. I don't think we can restrict to period because it's meant to execute on every query (basically it just does this: "are there queued invalidations? if so, let's invalidate before we launch archiving"). Since it's just for browser archiving I don't think we have to care too much about it.

Reply:
@diosmosis this could be still executed though when browser archiving is disabled right? And then because archive php is not triggered it might invalidate them? Might be good to test this

Reply:
Oh I see, I think in the old code we didn't invalidate in core:archive, so maybe it was all done here... I'll add the check for browser archiving being enabled.